tests/utils.py

```diff
@@ -48,13 +49,34 @@ def host_header(self) -> Tuple[bytes, bytes]:
 def http_proxy_server(proxy_host: str, proxy_port: int):
 
     proc = None
-    try:
-        command = ["pproxy", "-l", f"http://{proxy_host}:{proxy_port}/"]
-        proc = subprocess.Popen(command)
 
-        _wait_can_connect(proxy_host, proxy_port)
+    with create_proxy_block_file(["example.com"]) as block_file_name:
+        try:
+            command = [
+                "pproxy",
+                "-b",
+                block_file_name,
+                "-l",
+                f"http://{proxy_host}:{proxy_port}/",
+            ]
+            proc = subprocess.Popen(command)
+
+            _wait_can_connect(proxy_host, proxy_port)
+
+            yield b"http", proxy_host.encode(), proxy_port, b"/"
+        finally:
+            if proc is not None:
+                proc.kill()
+
+
+@contextlib.contextmanager
+def create_proxy_block_file(blocked_domains: List[str]):
```

Comment:
I'd suggest we add a small docstring here with any relevant docs material from `pproxy` (eg a link to relevant docs) describing what a "block file" is and what effect it has?

Reply:
I added some docs to `http_proxy_server` and `create_proxy_block_file`